spec/utils_spec.lua

```diff
@@ -793,4 +793,32 @@ describe("utils", function()
 			assert.is_true(result)
 		end)
 	end)
+
+	describe("booleanOrBooleanStringToBoolean", function()
+		it("should return a boolean as itself", function()
+			assert.is_true(utils.booleanOrBooleanStringToBoolean(true))
+			assert.is_false(utils.booleanOrBooleanStringToBoolean(false))
+		end)
+
+		it("should convert any casing of true or false to the analogous boolean value", function()
+			assert.is_true(utils.booleanOrBooleanStringToBoolean("true"))
+			assert.is_true(utils.booleanOrBooleanStringToBoolean("True"))
+			assert.is_true(utils.booleanOrBooleanStringToBoolean("TRUE"))
+			assert.is_true(utils.booleanOrBooleanStringToBoolean("tRuE"))
+			assert.is_false(utils.booleanOrBooleanStringToBoolean("false"))
+			assert.is_false(utils.booleanOrBooleanStringToBoolean("False"))
+			assert.is_false(utils.booleanOrBooleanStringToBoolean("FALSE"))
+			assert.is_false(utils.booleanOrBooleanStringToBoolean("fAlSe"))
+		end)
+
+		it("should not convert other truthy-like string values to boolean", function()
```

Comment:
recommend adding a test for nil, table, function, and thread (coroutine)

Reply:
51bbfe5